### PR TITLE
Add documentation Section to lucide.dev

### LINKED ISSUE #140: Create documentation section in the site.
I'm almost done with creating the react package, and for that I need to write some documentation as well. 
With that, maybe it is nice to centralize the documentation.

I got a bit inspired by the [ChakraUI website](https://chakra-ui.com/docs/getting-started), I think they have an great, simple and nice looking documentation site. I noticed they use their own UI + next.js, the same as we do. And they use enhanced markdown to generate documentation pages. (`.mdx` files), see their `[next.config.js](https://github.com/chakra-ui/chakra-ui/blob/develop/website/next.config.js)`.

So maybe we can use that as well.

### LINKED ISSUE #379: Documentation: Comparison with Feather
1. Small introduction why Lucide was born.
2. Why use lucide
3. Our goals:

The updated icons through the icons set, how we add more icons and what our goal is.
Our goals: 
- We're expanding the icon set as much as possible while keeping it nice-looking.
- Provide tools to implement lucide on the most well used libraries, frameworks and design programs.
- and more..

### LINKED ISSUE #394: React Native support through Expo
Perhaps an easier way to add support is via https://icons.expo.fyi/, rather than creating and mantain a new package

How to add the icon library to the expo project: https://github.com/expo/vector-icons

### LINKED ISSUE #430: show license info on website
It will be more clear and descriptive to show lucide package license on website.

### PR DESCRIPTION
**Pull request by ericfennis**
closes #140
closes #394
closes #430
closes #379

### CODE REVIEW

**Comment by locness3** [comment]
Deploy preview failed to build :/

**Comment by ericfennis** [comment]
@locness3 it's fixed!

**Comment by ericfennis** [comment]
Demo: 
image

**Comment by wojtekmaj** [comment]
Sorry, I can't see how exactly this resolves #394?

File: docs/comparison.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+title: Comparison
+---
+
+# Comparison
+
+## Lucide vs Feather Icons
+
+Lucide is a community-run fork of [Feather Icons](https://github.com/feathericons/feather).
+
+It began after growing disaffection of the [Feather Icons](https://github.com/feathericons/feather) project moderation. With over 300+ open issues and over 100+ open PRs, the Feather Icons project has been abandoned adn not maintained actively. This unfortunately means that hundreds of developers and designers wasted their time contributing to Feather Icons with no chance of PRs being accepted.
```
Comment (wojtekmaj) [inline docs/comparison.md:11]: Typo: "adn"

**Comment by ericfennis** [comment]
@wojtekmaj Sharp, sorry typo in the issue number, will reopen it again!

File: docs/comparison.md (hunk 1 above)
Comment (ericfennis) [reply docs/comparison.md:11]: tnx!

**Pull request merged**